File: my-app/src/app/common/matcheventdto.ts
import {ClubDto} from "./club-dto";
import {PlayerDto} from "./player-dto";
import {MatchDto} from "./match-dto";

export class Matcheventdto {
  id: number;
  minute: number;
  type: string;
  clubId: ClubDto;
  player: PlayerDto;
  assisted: PlayerDto;
  substitutePlayer: PlayerDto;
  isPenalty: boolean;
  isOwnGoal: boolean;
  redCard: boolean;
  yellowCard: boolean;
  description: string;
  matchId: MatchDto;

  constructor(
    id: number,
    minute: number,
    type: string,
    clubId: ClubDto,
    player: PlayerDto,
    assisted: PlayerDto,
    substitutePlayer: PlayerDto,
    isPenalty: boolean,
    isOwnGoal: boolean,
    redCard: boolean,
    yellowCard: boolean,
    description: string,
    matchId: MatchDto
  ) {
    this.id = id;
    this.minute = minute;
    this.type = type;
    this.clubId = clubId;
    this.player = player;
    this.assisted = assisted;
    this.substitutePlayer = substitutePlayer;
    this.isPenalty = isPenalty;
    this.isOwnGoal = isOwnGoal;
    this.redCard = redCard;
    this.yellowCard = yellowCard;
    this.description = description;
    this.matchId = matchId;
  }

}

File: FootballApp/src/main/java/com/football/dev/footballapp/dto/MatchEventDTO.java
package com.football.dev.footballapp.dto;

import com.football.dev.footballapp.models.enums.TypeStatus;

public record MatchEventDTO(
  Long id,
  Integer minute,
  TypeStatus type,
  ClubDto club_id,
  PlayerDto player,
  PlayerDto assisted,
  PlayerDto substitutePlayer,
  boolean isPenalty,
  boolean isOwnGoal,
  boolean redCard,
  boolean yellowCard,
  String description,
  MatchDTO matchId
) {}
